user-management/server/server.ts

```diff
@@ -0,0 +1,186 @@
+     *     Create a new signed token with `jwt.sign()`, using the payload and your TOKEN_SECRET
+     *       (see `hashKey` above).
+     *     Send the client a 200 response containing an object with 2 keys, "user" and "token",
+     *       where "user"'s value is the payload and "token"'s value is the token.
+     */
+
+    const sql = `
+    select "userId", "hashedPassword"
+    from "users"
+    where "username" = $1
+
+    `;
+    const params = [username];
+    const result = await db.query(sql, params);
+    const user = result.rows[0];
+    if (!user) {
+      throw new ClientError(401, 'invalid login');
+    }
+    const passwordMatches = await argon2.verify(user.hashedPassword, password);
+    if (!passwordMatches) {
+      throw new ClientError(401, 'invalid login');
+    }
+
+    // Create JWT
+    const payload = { userId: user.userId, username };
+    const token = jwt.sign(payload, process.env.TOKEN_SECRET);
+
+    // Send response
+    res.status(200).json({
+      user: payload,
+      token,
+    });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/todos', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "todos"
+        where "userId" = $1
+        order by "todoId"
+    `;
+    const result = await db.query(sql, [req.user?.userId]);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/todos', async (req, res, next) => {
+  try {
+    const { task, isCompleted = false } = req.body;
+    if (!task || typeof isCompleted !== 'boolean') {
+      throw new ClientError(400, 'task and isCompleted are required');
+    }
+    const sql = `
+      insert into "todos" ("userId", "task", "isCompleted")
+        values ($1, $2, $3)
+        returning *
+    `;
+    const params = [req.user?.userId, task, isCompleted];
+    const result = await db.query(sql, params);
+    const [todo] = result.rows;
+    res.status(201).json(todo);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/todos/:todoId', async (req, res, next) => {
+  try {
+    const todoId = Number(req.params.todoId);
+    if (!Number.isInteger(todoId) || todoId < 1) {
+      throw new ClientError(400, 'todoId must be a positive integer');
+    }
+    const { task, isCompleted } = req.body;
+    if (typeof isCompleted !== 'boolean') {
+      throw new ClientError(400, 'isCompleted (boolean) is required');
+    }
+    const sql = `
+      update "todos"
+        set "updatedAt" = now(),
+            "isCompleted" = $1,
+            "task" = $2
+        where "todoId" = $3 and "userId" = $4
+        returning *
+    `;
+    const params = [isCompleted, task, todoId, req.user?.userId];
+    const result = await db.query(sql, params);
+    const [todo] = result.rows;
+    if (!todo) {
+      throw new ClientError(404, `cannot find todo with todoId ${todoId}`);
+    }
+    res.json(todo);
+  } catch (err) {
+    next(err);
+  }
+});
+app.use(authMiddleware);
```

Comment:
You don't "use" the authMiddleware. Instead, you add it as the second parameter to all endpoint routes that need authorization.